src/ipc-handlers.ts

```diff
@@ -238,48 +241,63 @@ export async function connectWpcomSite(
 
 	const connections = userData.connectedWpcomSites[ currentUserId ];
 
-	sites.forEach( ( siteToAdd ) => {
-		const isAlreadyConnected = connections.some(
-			( conn ) => conn.id === siteToAdd.id && conn.localSiteId === localSiteId
-		);
+	list.forEach( ( { sites, localSiteId } ) => {
+		sites.forEach( ( siteToAdd ) => {
+			const isAlreadyConnected = connections.some(
+				( conn ) => conn.id === siteToAdd.id && conn.localSiteId === localSiteId
+			);
 
-		// Add the site if it's not already connected
-		if ( ! isAlreadyConnected ) {
-			connections.push( { ...siteToAdd, localSiteId } );
-		}
+			// Add the site if it's not already connected
+			if ( ! isAlreadyConnected ) {
+				connections.push( { ...siteToAdd, localSiteId } );
+			}
+		} );
 	} );
 
 	await saveUserData( userData );
 
-	return connections.filter( ( conn ) => conn.localSiteId === localSiteId );
+	return connections.filter( ( conn ) =>
+		list.some( ( { localSiteId } ) => conn.localSiteId === localSiteId )
+	);
 }
 
+type DataDisconnectWpcomSite = { siteIds: number[]; localSiteId: string };
+
 export async function disconnectWpcomSite(
 	event: IpcMainInvokeEvent,
-	siteIds: number[],
-	localSiteId: string
+	data: DataDisconnectWpcomSite | DataDisconnectWpcomSite[]
 ) {
+	const list = Array.isArray( data ) ? data : [ data ];
+
 	const userData = await loadUserData();
 	const currentUserId = userData.authToken?.id;
 
 	if ( ! currentUserId ) {
 		throw new Error( 'User not authenticated' );
 	}
 
-	const connections = userData.connectedWpcomSites?.[ currentUserId ] || [];
+	const connectedWpcomSites = userData.connectedWpcomSites;
 
-	const updatedConnections = connections.filter(
-		( conn ) => ! ( siteIds.includes( conn.id ) && conn.localSiteId === localSiteId )
-	);
+	// Totally unreal case, added it to make types below more clear and if this error happens, we definitely have something wrong
```

Comment:
I added this comment and condition to make code below more obvious and avoid TS errors/warnings. Otherwise we need to do optional chaining and add extra conditions below, which are just confusing and unnecessary.
Initially, I tried to go that way, but it was not looking nice and even confused myself during development.